### PR TITLE
engage: init at 0.1.2

### PR DESCRIPTION
**Pull request by CobaltCause**
###### Description of changes

Packages a handy tool I wrote primarily for writing and running CI pipelines.

Let me know if there's a more appropriate place to put this package.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/engage/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "v${version}";
+    hash = "sha256-7zLFgTLeAIaMMoj0iThH/5UhnV9OUGe9CVwbbShCieo=";
+  };
+
+  cargoHash = "sha256-+4uqC0VoBSmkS9hYC1lzWeJmK873slZT04TljHPE+Eo=";
+
+  meta = with lib; {
+    description = "A task runner with DAG-based parallelism";
+    homepage = "https://or.computer.surgery/charles/engage";
+    changelog = "https://or.computer.surgery/charles/engage/-/blob/main/CHANGELOG.md";
```
Comment (0x4A6F) [inline pkgs/development/tools/engage/default.nix:26]:
```diff
-    changelog = "https://or.computer.surgery/charles/engage/-/blob/main/CHANGELOG.md";
+    changelog = "https://or.computer.surgery/charles/engage/-/blob/v${version}/CHANGELOG.md";
```

**Review by 0x4A6F — CHANGES_REQUESTED** [review]

**Comment by 0x4A6F** [comment]
Result of `nixpkgs-review pr 217439` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>engage</li>
  </ul>
</details>

**Comment by 0x4A6F** [comment]
Result of `nixpkgs-review pr 217439` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>engage</li>
  </ul>
</details>

**Comment by 0x4A6F** [comment]
Result of `nixpkgs-review pr 217439` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>engage</li>
  </ul>
</details>

**Comment by 0x4A6F** [comment]
Result of `nixpkgs-review pr 217439` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>engage</li>
  </ul>
</details>

**Review by 0x4A6F — APPROVED** [review]

File: pkgs/development/tools/engage/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, rustPlatform
+, fetchgit
+}:
+
+let
+  pname = "engage";
+  version = "0.1.2";
+in
+
```
Comment (AndersonTorres) [inline pkgs/development/tools/engage/default.nix:10]:
```suggestion
```

File: pkgs/development/tools/engage/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  cargoHash = "sha256-+4uqC0VoBSmkS9hYC1lzWeJmK873slZT04TljHPE+Eo=";
+
+  meta = with lib; {
+    description = "A task runner with DAG-based parallelism";
+    homepage = "https://or.computer.surgery/charles/engage";
+    changelog = "https://or.computer.surgery/charles/engage/-/blob/v${version}/CHANGELOG.md";
+    license = with licenses; [ asl20 mit ];
+    maintainers = with maintainers; [ CobaltCause ];
```
Comment (AndersonTorres) [inline pkgs/development/tools/engage/default.nix:28]:
```suggestion
  meta = {
    description = "A task runner with DAG-based parallelism";
    homepage = "https://or.computer.surgery/charles/engage";
    changelog = "https://or.computer.surgery/charles/engage/-/blob/v${version}/CHANGELOG.md";
    license = with lib.licenses; [ asl20 mit ];
    maintainers = with lib.maintainers; [ CobaltCause ];
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -4550,6 +4550,8 @@ with pkgs;
 
   enca = callPackage ../tools/text/enca { };
 
+  engage = callPackage ../development/tools/engage { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:4553]: This is not restricted to development. I believe it should be put on `tools/misc`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Pull request merged**